spec/command/terraform/generate_spec.rb

```diff
@@ -9,22 +9,98 @@
 TERRAFORM_CONFIG_DIR_PATH = GENERATOR_PLAYGROUND_PATH.join("terraform")
 
 describe Command::Terraform::Generate do
-  before do
-    FileUtils.rm_r(GENERATOR_PLAYGROUND_PATH) if Dir.exist?(GENERATOR_PLAYGROUND_PATH)
-    FileUtils.mkdir_p GENERATOR_PLAYGROUND_PATH
+  subject(:result) { run_cpflow_command(described_class::SUBCOMMAND_NAME, described_class::NAME, *options) }
+
+  let!(:app) { dummy_test_app }
+  let(:options) { ["-a", app] }
 
+  before do
+    FileUtils.rm_rf(GENERATOR_PLAYGROUND_PATH)
+    FileUtils.mkdir_p(GENERATOR_PLAYGROUND_PATH)
     allow(Cpflow).to receive(:root_path).and_return(GENERATOR_PLAYGROUND_PATH)
   end
 
   after do
-    FileUtils.rm_r GENERATOR_PLAYGROUND_PATH
+    FileUtils.rm_rf GENERATOR_PLAYGROUND_PATH
+  end
+
+  it "generates terraform config files", :aggregate_failures do
+    config_file_paths.each { |config_file_path| expect(config_file_path).not_to exist }
+
+    expect(result[:status]).to eq(0)
+
+    expect(config_file_paths).to all(exist)
   end
 
-  it "generates terraform config files" do
-    providers_config_file_path = TERRAFORM_CONFIG_DIR_PATH.join("providers.tf")
+  context "when templates folder is empty" do
+    let(:template_dir) { "non-existing-folder" }
+
+    before do
+      # rubocop:disable RSpec/AnyInstance
+      allow_any_instance_of(TemplateParser).to receive(:template_dir).and_return(template_dir)
+      # rubocop:enable RSpec/AnyInstance
```

Comment:
Nit: You could add `# rubocop:disable RSpec/AnyInstance` at the end line 40 instead.